### PR TITLE
feat: zil support on polygon

### PR DESCRIPTION
**Pull request by enaqx**
Sample code to test

```swift
import Foundation
import UnstoppableDomainsResolution

func main() {
    do {
        var resolution = try Resolution(configs: Configurations(
            uns: UnsLocations(
                layer1: NamingServiceConfig(
                    providerUrl: "https://eth-mainnet.alchemyapi.io/v2/_BDuTLPgioYxULIE5cGq3wivWAJborcM",
                    network: "mainnet"),
                layer2: NamingServiceConfig(
                    providerUrl: "https://polygon-mumbai.g.alchemy.com/v2/ymbY17ik_HyGfXnPWxBAGhuZE7MwtErX",
                    network: "polygon-mumbai"),
                zlayer: NamingServiceConfig(
                    providerUrl: "https://api.zilliqa.com",
                    network: "mainnet")
            )
        ));
                
        // zil domains resolved on L2
        let testDomain = "uns-devtest-testdomain303030.zil" 
                
        resolution.owner(domain: testDomain) { (result) in
            switch result {
            case .success(let returnValue):
                // 0x499dD6D875787869670900a2130223D85d4F6Aa7
                print(returnValue)
            case .failure(let error):
                print("Expected Address, but got \(error)")
            }}
        
    } catch {
        print ("Init of Resolution instance with default parameters failed...")
        return
    }
}

main()

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -72,30 +57,21 @@ guard let resolution = try? Resolution() else {
...
-Library can offer three naming services at the moment:
-
-* `uns` resolves `.crypto` ,  `.coin`,  `.wallet`,  `.bitcoin`,  `.blockchain`, `.x`, `.888`, `.nft`, `.dao` domains,
-* `zns` resolves `.zil` domains
-
-By default, each of them is using the mainnet network via infura provider.
-Unstoppable domains are using the infura key with no restriction for UNS.  
-
-You can update each naming service separately
+Library support three networks at the moment Ethereum, Polygon and Zilliqa. You can update each network separately.
```
Comment (rommex) [inline README.md:60]: typo: must be `supports`

File: Sources/UnstoppableDomainsResolution/NamingServices/UNS.swift [hunk 2]
```diff
@@ -122,9 +139,9 @@ internal class UNS: CommonNamingService, NamingService {
         let l2Result = Utillities.getLayerResult(from: results, for: .layer2)
         let l1Result = Utillities.getLayerResult(from: results, for: .layer1)
 
-        for (domain, (l2owner, l1owner)) in zip(domains, zip(l2Result, l1Result)) {
-            owners[domain] = l2owner == nil ? l1owner : l2owner
-        }
+//        for (domain, (l2owner, l1owner)) in zip(domains, zip(l2Result, l1Result)) {
```
Comment (rommex) [inline Sources/UnstoppableDomainsResolution/NamingServices/UNS.swift:142]: Dead code?

File: Sources/UnstoppableDomainsResolution/Resolution.swift [hunk 3]
```diff
@@ -94,15 +94,23 @@ public class Resolution {
...
-        DispatchQueue.global(qos: .utility).async { [weak self] in
-            do {
-                if let preparedDomain = try self?.prepare(domain: domain),
-                    let result = try self?.getServiceOf(domain: preparedDomain).owner(domain: preparedDomain) {
-                    completion(.success(result))
-                }
-            } catch {
-                self?.catchError(error, completion: completion)
-            }
+//        DispatchQueue.global(qos: .utility).async { [weak self] in
```
Comment (rommex) [inline Sources/UnstoppableDomainsResolution/Resolution.swift:97]: Dead code?

File: Sources/UnstoppableDomainsResolution/Resolution.swift [hunk 4]
```diff
@@ -414,9 +413,9 @@ public class Resolution {
 
     /// This returns the correct naming service based on the `domain` asked for
     private func getServiceOf(domain: String) throws -> NamingService {
-        if domain.hasSuffix(".zil") {
-            return try self.findService(name: .zns)
-        }
+//        if domain.hasSuffix(".zil") {
```
Comment (rommex) [inline Sources/UnstoppableDomainsResolution/Resolution.swift:416]: Dead code?

**Review by rommex — APPROVED** [review]
LGTM with minor comments

File: Sources/UnstoppableDomainsResolution/NamingServices/UNS.swift [hunk 5]
```diff
@@ -35,65 +41,75 @@ internal class UNS: CommonNamingService, NamingService {
...
         )
     }
 
     func locations(domains: [String]) throws -> [String: Location] {
         let results = try asyncResolver.resolve(
-            l1func: self.layer1.locations(domains: domains),
-            l2func: self.layer2.locations(domains: domains)
+            listOfFunc: [{try self.layer1.locations(domains: domains)},
+                         {try self.layer2.locations(domains: domains)},
+                         {try self.zlayer.locations(domains: domains)}]
```
Comment (DancingAxolotl) [inline Sources/UnstoppableDomainsResolution/NamingServices/UNS.swift:110]: ZNS locations method throws a "methodNotSupported" error. We should either handle the error or return something for ZNS locations. We can probably check domain owner and return a partial `Location` struct

**Review by DancingAxolotl — CHANGES_REQUESTED** [review]
Good work! 
I left some comments for ZNS functions that should be addressed, but otherwise the PR looks good.

**Comment by enaqx** [comment]
Updated, please take a look

**Pull request merged**